### PR TITLE
Remove user fee

### PR DESCRIPTION
**Pull request by sunce86**
# Description
Fixes https://github.com/cowprotocol/services/issues/2492

Related gnosis-solvers PR: https://github.com/gnosis/solvers/pull/9

Will be merged next week - apparently we don't have enough solvers ready for this breaking change yet!

# Changes
Removed field "user_fee" from auction endpoint - breaking API change.
Removed field "fee_amount" from solvers api - also breaking API change

## How to test
Existing (updated) tests.

### CODE REVIEW

**Review by squadgazzz — APPROVED** [review]
LG

File: crates/driver/src/domain/competition/solution/trade.rs
Comment (MartinquaXD) [inline crates/driver/src/domain/competition/solution/trade.rs:92]:
```suggestion
                // Orders with static fees are no longer used, except for quoting purposes, when
```

File: crates/driver/src/boundary/settlement.rs
Comment (MartinquaXD) [inline crates/driver/src/boundary/settlement.rs:265]: I think commenting here that `fee` is guaranteed to be `0` and therefore omitted from the remaining source code but is still required to correctly encode the order here would be useful.
Actually given that solvers can encode their own solutions in the future we probably need to mention this somewhere in the docs.

**Review by MartinquaXD — APPROVED** [review]
Really nice.
Also needs a change in https://github.com/gnosis/solvers.
Let's hope none of the external solvers break because of this. :crossed_fingers:
Should we maybe run this commit in staging before merging to see if some of the solvers will run into issues because of it.
cc @harisang to communicate this change to solvers.

File: crates/solvers-dto/src/auction.rs
Comment (fleupold) [inline crates/solvers-dto/src/auction.rs:38]: How confident are we that this breaking change is not going to shoot us in the foot? I think it would be ok to leave this value on the API level but simply always set it to 0 in code (and document that it's deprecated and always 0 in the API spec). For our internal models (domain, etc), we would want to remove this field however.

The frontend team would like to re-vamp our API and I think it would be ok to defer breaking changes thereto.

File: crates/driver/src/domain/competition/solution/trade.rs
Comment (fleupold) [inline crates/driver/src/domain/competition/solution/trade.rs:93]: Is there anything blocking #2543 ? If not could we simply implement this now to combine the cleanups (I'm not sure there will be time scheduled for the follow up cleanup)

**Review by fleupold — COMMENTED** [review]
Code looks good overall, I'm just somewhat worried about breaking the API for clients and wonder if we can decouple our internal representation (no more user fee) and the external representation (hardcode user fee to 0) in a way that could lower this risk?

File: crates/solvers-dto/src/auction.rs
Comment (sunce86) [reply crates/solvers-dto/src/auction.rs:38]: The mentioned field is part of the solvers API. Not sure what frontend has with it? If you got confused and referred to the `model` data struct than yes, we still want to keep `fee_amount` there.
Solvers API should be kept clean, this moment is good enough to make this change as any other IMO (actually, would have been better if we made this change before solvers moved to this API 😄  buy hey...)

Comment (fleupold) [reply crates/solvers-dto/src/auction.rs:38]: Sorry, wrong file (I wanted to comment on the one above: crates/orderbook/src/dto/order.rs), there the field is `user_fee`. Just want to make sure only our frontend depends on this field and can live without it.

Comment (sunce86) [reply crates/solvers-dto/src/auction.rs:38]: This is a similar change we had few weeks ago with `solver_fee`: https://github.com/cowprotocol/services/pull/2272 No major problems arose except shadow competition was down and alerter being down AFAIR.

Comment (sunce86) [reply crates/solvers-dto/src/auction.rs:38]: This time I would expect only shadow to be down because alerter is not deserializing `user_fee`.

**Review by fleupold — APPROVED** [review]

File: crates/driver/src/boundary/settlement.rs
Comment (sunce86) [reply crates/driver/src/boundary/settlement.rs:268]: Added comment in the code.

File: crates/driver/src/domain/competition/solution/trade.rs
Comment (sunce86) [reply crates/driver/src/domain/competition/solution/trade.rs:93]: Started working on it here: https://github.com/cowprotocol/services/pull/2590
Do you think #2543 should block merging this PR?

**Review by Mateo-mro — APPROVED** [review]

**Pull request merged**
